### ISSUE TITLE
Auth logging not working

### ISSUE THREAD

**Issue by ayamit** [RainLoop/rainloop-webmail#665]
I have a new install of RainLoop v 1.9.0.327 (Community Edition) running fine except for auth logging. Regular logging works fine, but I cannot get the authentication logging to work (intended to use with Fail2Ban).

Here is the log entry from application.ini:

```
[logs]
; Enable logging
enable = On

; Logs entire request only if error occured (php requred)
write_on_error_only = On

; Logs entire request only if php error occured
write_on_php_error_only = On

; Logs entire request only if request timeout (in seconds) occured.
write_on_timeout_only = 0

; Required for development purposes only.
; Disabling this option is not recommended.
hide_passwords = On
time_offset = 0

; Log filename.
; For security reasons, some characters are removed from filename.
; Allows for pattern-based folder creation (see examples below).
; 
; Patterns:
...

; Enable auth logging in a separate file (for fail2ban)
auth_logging = On
auth_logging_filename = "auth.txt"
auth_logging_format = "Auth failed: ip={request:ip} user={imap:login} host={imap:host} port={imap:port}"
```

All I changed was to enable, with Error Only, and change the log name. However, nothing is being written to the auth log, neither on successful or failed logins. I get full details in the regular log, but zero activity in the auth log. I tried creating the auth log file, with write permissions, but still no logging to it.

**Comment by RainLoop** [comment]
Hmmm, I've copied your settings and everything work correctly.
Any errors in the apache/php/nginx error logs?

**Comment by ayamit** [comment]
There is nothing in the Apache error log or the Linux syslog. I can't find any sign of an error, just a lack of auth logging. Is there a specific debugging mode I can use to learn more?

**Comment by ayamit** [comment]
To clarify, I am running RainLoop in Ubuntu 14.04 using PHP 5 as an Aapche module. Apache user is www-data and that is the owner of the whole RainLoop directory.

**Comment by RainLoop** [comment]
Have you installed Selinux?

**Comment by ayamit** [comment]
No Selinux, but AppArmor is installed.

What makes me curious is how one of the logs (regular) can work while the other (auth) does not. What difference is there is the code allowing one to fail?

Also, the regular log is fairly verbose; is there anything I should look for there to indicate a write error or some other cause of failure?

**Comment by RainLoop** [comment]
I do not know.
Code have a similar logic.

**Comment by ayamit** [comment]
Okay, found the problem: not in RainLoop or my config: the IMAP server I'm connecting to is behind an Nginx reverse proxy, and something about the failure message Nginx is giving is not registering for logging in RainLoop. When I connect RainLoop straight to a Dovecot IMAP server, no proxy, auth failures are logged. The Nginx proxy works fine for successfully logging in with RainLoop, and every other mail client I've used, so it's only on failure that RainLoop doesn't recognize the response for what it is.

Has anyone else seen a problem like this?

**Comment by ayamit** [comment]
Or, what is the response pattern RainLoop looks for for auth failures? I might be able to modify Nginx to give the right kind of response so RainLoop will log it.

**Comment by ikolubr** [comment]
Same problem here! Version 1.9.2.361
Nothing on auth.log
regular log works just fine.
no errors on rsyslog or apache.

connecting directo to dovecot on same machine. everything else works just flawlessly but the auth.log, and this piece is very important for us, as you can imagine!

Thank you,
Igor

**Comment by RainLoop** [comment]
Please, try the latest version (1.9.3.363)

**Comment by ayamit** [comment]
I tried to the new version without success. Both manually upgrading and a clean install. I enabled auth logging in application.ini and was able to successfully login with the correct password. On failure there is no log written. The file isn't even created. The regular log is successfully written.

One possibly relevant line in the regular log is:
AJAX[DATA]: {"Action":"Login","Result":false,"ErrorCode":102,"ErrorMessage":"","ErrorMessageAdditional":"","Time":2316}
